consensus/hotstuff/committees/leader/leader_selection_test.go

```diff
@@ -19,6 +20,47 @@ var someSeed = []uint8{0x6A, 0x23, 0x41, 0xB7, 0x80, 0xE1, 0x64, 0x59,
 	0x66, 0x53, 0x41, 0xB7, 0x80, 0xE1, 0x64, 0x51,
 	0xAA, 0x53, 0x40, 0xB7, 0x80, 0xE4, 0x64, 0x50}
 
+// We test that leader selection works for a committee of size one
+func TestSingleConsensusNode(t *testing.T) {
+	identity := unittest.IdentityFixture(unittest.WithStake(8))
+	selection, err := ComputeLeaderSelectionFromSeed(0, someSeed, 10, []*flow.Identity{identity})
+	require.NoError(t, err)
+	for i := uint64(0); i < 10; i++ {
+		leaderID, err := selection.LeaderForView(i)
+		require.NoError(t, err)
+		require.Equal(t, identity.NodeID, leaderID)
+	}
+}
+
+// compare this binary search method with sort.Search()
+func TestBsearchVSsortSearch(t *testing.T) {
+	stakes := []uint64{1, 2, 3, 4, 5, 6, 7, 9, 12, 21, 32}
+	stakes2 := []int{1, 2, 3, 4, 5, 6, 7, 9, 12, 21, 32}
+	var sum uint64
+	var sum2 int
+	sums := make([]uint64, 0)
+	sums2 := make([]int, 0)
+	for i := 0; i < len(stakes); i++ {
+		sum += stakes[i]
+		sum2 += stakes2[i]
+		sums = append(sums, sum)
+		sums2 = append(sums2, sum2)
+	}
+	sel := make([]int, 0, 10)
+	for i := 0; i < 10; i++ {
+		index := binarySearch(uint64(i), sums)
+		sel = append(sel, index)
+	}
+
+	sel2 := make([]int, 0, 10)
+	for i2 := 1; i2 < 11; i2++ {
```

Comment:
binarySearch() works by finding the first element in list that is strictly bigger.  It doesn't work the same way as sort.SearchInts(), so the initial values are different.  Maybe the name should change, any suggestion ? like 'binarySearchStrictlyBigger()'

Reply:
I do think a name change might be helpful, since search does usually imply looking for the value. I know it's in the comments, but might be worth a name change as well

Reply:
done